runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -34,10 +34,20 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
+
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
    {
-   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill && !(!comp->getOption(TR_DisableVectorAPIExpansion) && comp->getMethodSymbol()->hasVectorAPI());
+   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear")) == NULL;
+   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion))
+      {
+      TR_VirtualGuard *guardInfo = comp->findVirtualGuardInfo(virtualGuardNode);
+      if (guardInfo != NULL && guardInfo->isInlineGuard())
+         {
+         char* methodSig = guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()->classNameChars();
```

Comment:
I think `methodSig` is a misleading name for this string, which is actually a class name